meinberlin/apps/budgeting/assets/SwitchButton.jsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react'
+
+export default class SwitchButton extends React.Component {
+  constructor (props) {
+    super(props)
```

Comment:
@phillimorland i think that is a very good idea to have a switch component, where you can change the text and its appearance!

yes we have to handle state here so the class notation is fine and still very readable. It should be possible to write it as functional component, but lets keep this as its working and its readable.

Additional note (but in this case not useful i would say):

A different way to make this component type more like an html element would be using `props.children`:

```
<SwitchButton>
    {django.gettext('Voted')}
</SwitchButton>
```
Or to make the different states for `on` and `off` it would be:
```
<SwitchButton>
    { {
            on: django.gettext('Voted'),
            off: django.gettext('Give my vote')
    }}
</SwitchButton>

But i think because we want to have two states editable here the props.children approach is too complicated to read.

Reply:
coolio! will leave like it is and will hook it up in anothe pull request so I guess this is good to go, I removed the voter code form as we'll now do that in django